model/common/src/icon4py/model/common/states/factory.py

```diff
@@ -445,8 +446,8 @@ def _get_offset_providers(self, grid: icon_grid.IconGrid) -> dict[str, gtx.Field
                 horizontal_offsets = {
                     k: v
                     for k, v in grid.offset_providers.items()
-                    if isinstance(v, gtx.NeighborTableOffsetProvider)
-                    and v.origin_axis.kind == gtx.DimensionKind.HORIZONTAL
+                    if isinstance(v, gtx.Connectivity)
```

Comment:
We should be able to remove this workaround with [PR 1746](https://github.com/GridTools/gt4py/pull/1746)...